### PR TITLE
fix: rename struct and deprecated the old name

### PR DESCRIPTION
**Pull request by aymanbagabas**
`SSHKeyPair` is way too long and unnecessary

### CODE REVIEW

File: keygen.go [hunk 1]
```diff
@@ -95,43 +99,43 @@ type SSHKeyPair struct {
...
+func (s KeyPair) privateKeyPath() string {
 	return s.path
 }
 
-func (s SSHKeyPair) publicKeyPath() string {
+func (s KeyPair) publicKeyPath() string {
 	return s.privateKeyPath() + ".pub"
 }
 
 // Option is a functional option for SSHKeyPair.
```
Comment (bashbunni) [inline keygen.go:110]: Should this comment be KeyPair instead of the deprecated SSHKeyPair?

Comment (aymanbagabas) [inline keygen.go:110]:
```suggestion
// Option is a functional option for KeyPair.
```

**Comment by bashbunni** [comment]
Feel free to undo my commit if it's not helpful, just renamed the other references of `SSHKeyPair`

**Comment by aymanbagabas** [comment]
> Feel free to undo my commit if it's not helpful, just renamed the other references of `SSHKeyPair`

Thanks!

**Pull request merged**